aws/src/test/java/org/apache/iceberg/aws/TestAwsClientFactories.java

```diff
@@ -56,6 +56,47 @@ public void testLoadDefault() {
         .isInstanceOf(AwsClientFactories.DefaultAwsClientFactory.class);
   }
 
+  @Test
+  public void testS3AsyncClientCrtEnabledTrue() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(S3FileIOProperties.ACCESS_KEY_ID, "keyId");
+    properties.put(S3FileIOProperties.SECRET_ACCESS_KEY, "accessKey");
+    properties.put(S3FileIOProperties.S3_CRT_ENABLED, "true");
+    properties.put(AwsClientProperties.CLIENT_REGION, "us-east-1");
+
+    AwsClientFactory clientFactory = AwsClientFactories.from(properties);
+    S3AsyncClient s3AsyncClient = clientFactory.s3Async();
+    assertThat(s3AsyncClient.getClass().getName())
```

Comment:
I would probably just simplify this a bit and also use an instanceof check instead of comparing the actual class name:

```
assertThat(
            AwsClientFactories.from(
                    ImmutableMap.of(
                        S3FileIOProperties.ACCESS_KEY_ID,
                        "keyId",
                        S3FileIOProperties.SECRET_ACCESS_KEY,
                        "accessKey",
                        S3FileIOProperties.S3_CRT_ENABLED,
                        "true",
                        AwsClientProperties.CLIENT_REGION,
                        "us-east-1"))
                .s3Async())
        .isInstanceOf(DefaultS3CrtAsyncClient.class);
```

Reply:
Ack.  I have implemented this change.

For posterity:

"DefaultS3CrtAsyncClient" is a public class

"DefaultS3AsyncClient" is not a public class, so it appears that I need to check the string name of the class.